skimage/filters/_multiotsu.pyx

```diff
@@ -6,101 +6,426 @@ import numpy as np
+    thresh_idx : int
+        Current index in thresh_indices.
+    nbins : int
         number of bins used in the histogram
-    divisions : int
+    thresh_count : int
         Number of divisions required to generate the desired classes.
-    depth : int
-        Controls the iterations the algorithm had, expanding the interval
-        when _find_best_rec() is called.
+    sigma_max : float
+        Current maximum variance between classes.
+    current_indices: (thresh_count, ) 1D array
+        Current evalueted threshold indices.
+    thresh_indices : array
+        The indices of thresholds maximizing the variance between
+        classes.
+
+    Returns
+    -------
     max_sigma : float
         Maximum variance between classes.
-    aux_thresh : array
-        Values for multi-Otsu threshold.
+
+    Notes
+    -----
+    For any candidate current_indices {t_0, ..., t_n}, sigma equals
+    var_btwcls[0, t_0] + var_btwcls[t_0+1, t_1] +
+    ... + var_btwcls[t_(i-1) + 1, t_i] + ... var_btwcls[t_n, nbins-1]
+
+    References
+    ----------
+    .. [1] Liao, P-S., Chen, T-S. and Chung, P-C., "A fast algorithm for
+           multilevel thresholding", Journal of Information Science and
+           Engineering 17 (5): 713-727, 2001. Available at:
+           <http://ftp.iis.sinica.edu.tw/JISE/2001/200109_01.pdf>
+           :DOI:`10.6688/JISE.2001.17.5.1`
+
+    """
+    cdef cnp.intp_t idx
+    cdef float sigma
+
+    if thresh_idx < thresh_count:
+
+        for idx in range(hist_idx, nbins - thresh_count + thresh_idx):
+            current_indices[thresh_idx] = idx
+            sigma_max = _set_thresh_indices_lut(var_btwcls, hist_idx=idx + 1,
+                                                thresh_idx=thresh_idx + 1,
+                                                nbins=nbins,
+                                                thresh_count=thresh_count,
+                                                sigma_max=sigma_max,
+                                                current_indices=current_indices,
+                                                thresh_indices=thresh_indices)
+
+    else:
+
+        sigma = (_get_var_btwclas_lut(var_btwcls, 0, current_indices[0], nbins)
+                 + _get_var_btwclas_lut(var_btwcls,
+                                        current_indices[thresh_count - 1] + 1,
+                                        nbins - 1, nbins))
+        for idx in range(thresh_count - 1):
+            sigma += _get_var_btwclas_lut(var_btwcls, current_indices[idx] + 1,
+                                          current_indices[idx + 1], nbins)
+
+        if sigma > sigma_max:
+            sigma_max = sigma
+            thresh_indices[:] = current_indices[:]
+
+    return sigma_max
+
+
+def _get_multiotsu_thresh_indices(float [::1] prob, Py_ssize_t thresh_count):
+    """Finds the indices of Otsu thresholds according to the values
+    occurence probabilities.
+
+    Parameters
+    ----------
+    prob: 1D array
```

Comment:
`prob : array`
You got the idea, right? :) Please, `var_name<space>:<space>datatype`. 1D arrays would be only `array`.